### PR TITLE
[FEATURE] MS-18: Companies refactor

### PR DESCRIPTION
**Pull request by Sabrucita**
MS-18: refactor companies, change extensions to jsx, delete old components

### CODE REVIEW

File: src/Components/Sessions/Form/index.jsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ function Form() {
           element="select"
           resource="candidates"
           name="candidate"
-          objectProperty="idCandidate"
+          //objectProperty="idCandidate"
```
Comment (BaltazarCortina) [inline src/Components/Sessions/Form/index.jsx]: Remove comment

File: src/Components/Companies/Form/index.jsx [hunk 2]
```diff
@@ -1,265 +1,253 @@
...
-  const onSubmit = (e) => {
+  const submitForm = (e) => {
     e.preventDefault();
-    if (clientId) {
-      //FUNCTION FOR UPDATING A COMPANY
-      fetch(`${process.env.REACT_APP_API}/companies/${clientId}`, {
-        method: 'PUT',
-        mode: 'cors',
+    setDisableProperty(true);
+    console.log(formData);
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx]: Remove console.log

File: src/Components/Companies/Form/index.jsx [hunk 3]
```diff
@@ -1,265 +1,253 @@
...
-    <div className={styles.container}>
-      <Modal show={showModal} closeModal={closeModal} action={lastAction} />
-      <form onSubmit={onSubmit}>
-        <h2>Form</h2>
-        <label id="fullname">Full Name</label>
-        <input
-          type="text"
-          id="fullname"
+    <div>
+      {operation === 'create' ? <h2>Create Company</h2> : <h2>Edit Company</h2>}
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx]: Replace h2 for h1

File: src/Components/Companies/List/list.module.css [hunk 4]
```diff
@@ -93,6 +33,10 @@ th {
   justify-content: center;
 }
 
+.action:hover{
+  cursor: pointer;
+}
+
 .action a:nth-of-type(3) {
   background-color: rgb(248, 83, 83);
   color: rgb(20, 20, 20);
```
Comment (BaltazarCortina) [inline src/Components/Companies/List/list.module.css:42]: This is not necessary

File: src/Components/Companies/Form/index.jsx [hunk 5]
```diff
@@ -1,265 +1,252 @@
...
+          updateData={updateForm}
         />
-        <label id="zipCode">Zip Code</label>
-        <input
-          type="number"
-          id="zipCode"
+        <Fieldset
+          update={id ? true : false}
+          currentValue={formData.zipCode}
+          element="input"
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx:176]: Should be type="number"

File: src/Components/Companies/Form/index.jsx [hunk 6]
```diff
@@ -1,265 +1,252 @@
...
+          updateData={updateForm}
         />
-        <label id="phone">Phone</label>
-        <input
-          type="text"
-          id="phone"
+        <Fieldset
+          update={id ? true : false}
+          currentValue={formData.phone}
+          element="input"
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx:186]: Should be type="number"

File: src/Components/Companies/Form/index.jsx [hunk 7]
```diff
@@ -1,265 +1,252 @@
...
+          updateData={updateForm}
         />
-        <label id="email">Email</label>
-        <input
-          type="text"
-          id="email"
+        <Fieldset
+          update={id ? true : false}
+          currentValue={formData.email}
+          element="input"
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx:196]: Should be type="email"

File: src/Components/Companies/Form/index.jsx [hunk 8]
```diff
@@ -1,265 +1,252 @@
...
+          updateData={updateForm}
         />
-        <label id="contactPhone">Contact Phone</label>
-        <input
-          type="text"
-          id="contactPhone"
+        <Fieldset
+          update={id ? true : false}
+          currentValue={formData.contactPhone}
+          element="input"
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx:226]: Should be type="number"

File: src/Components/Companies/Form/index.jsx [hunk 9]
```diff
@@ -1,265 +1,252 @@
...
+        }
       })
-        .then((response) => response.json())
-        .then((response) => {
-          console.log(response);
-          setLastAction('update');
+        .then(async (res) => {
+          const data = await res.json();
           setShowModal(true);
+          if (data) {
```
Comment (BaltazarCortina) [inline src/Components/Companies/Form/index.jsx:59]: if (data.status === 201) => company created
else => show error

**Review by EugenioPonzone — APPROVED** [review]

**Review by BaltazarCortina — APPROVED** [review]

**Review by juanandres7698 — APPROVED** [review]

**Review by JamirMansilla — APPROVED** [review]
Approved! 🏎️

**Pull request merged**